napari/components/viewer_model.py

```diff
@@ -364,6 +365,32 @@ def to_svg(self, file=None, view_box=None):
 
         return svg
 
+    def add_layer_by_type(self, layer_type, data, **kwargs):
```

Comment:
How do people feel about this being a private method `_ add_layer_by_type` or making this `add_layer` and rename our current a `add_layer` something like `_add_layer`, or do we like it as proposed @jni?

I know this will be useful when doing things for us internally like loading a `napari.zarr` file (see #511) or for the pub/sub syncing with dask (see #686), but I'm not sure we want this to be a major usage mode for users (thinking about what the doc string says etc) which is why I lean towards making it private

Reply:
(a) I don't understand why there is an Enum being used instead of the classes directly, (b) I definitely think this should be private, and (c) if we use the classes, then this method becomes trivial:

```python
def _add_layer(self, layer_type, data, **kwargs):
    return self.add_layer(layer_type(data, **kwargs))
```

... and in general I prefer that public-facing functions (and most other functions) don't have free `**kwargs` arguments, as they are much more difficult to introspect.

Reply:
Does add layer work for every layer type? There seems to be a different method for each layer type (add_volume, add_surface, etc) each doing a different set of things

Reply:
No, @shanaxel42, I wouldn't use `add_layer` here as there is a bunch of custom processing inside some of our add_* methods